guillotina/contrib/dbusers/services/users.py

```diff
@@ -0,0 +1,139 @@
+
+
+class BaseUser(Service):
+    async def get_user(self):
+        user_id = self.request.matchdict["user"]
+        try:
+            user = await navigate_to(self.context, "users/{}".format(user_id))
+        except KeyError:
+            user = None
+
+        if user is None:
+            raise HTTPNotFound(content={"reason": f"User {user_id} not found"})
+        return user
+
+
+@configure.service(
+    context=IContainer,
+    name="@users/{user}",
+    method="GET",
+    permission="guillotina.ManageUsers",
+    responses={
+        "200": {
+            "description": "User data",
+            # TODO: add response content schema here
+        },
+        "404": {"description": "User not found"},
+    },
+    summary="Get user data",
+    allow_access=True,
+)
+class GetUsers(BaseUser):
+    async def __call__(self):
+        user = await self.get_user()
+        serializer = get_multi_adapter((user, self.request), IResourceSerializeToJsonSummary)
+        result = await serializer()
+        return result
+
+
+@configure.service(
+    context=IContainer,
+    name="@users/{user}",
+    method="PATCH",
+    permission="guillotina.ManageUsers",
+    responses={
+        "204": {"description": "User successfully modified"},
+        "404": {"description": "User not found"},
+    },
+    summary="Modify user data",
+    allow_access=True,
+)
+class PatchUser(BaseUser):
+    async def __call__(self):
+        user = await self.get_user()
+        alsoProvides(self.request, IPATCH)
+        view = DefaultPATCH(user, self.request)
+        return await view()
+
+
+@configure.service(
+    context=IContainer,
+    name="@users/{user}",
+    method="DELETE",
+    permission="guillotina.ManageUsers",
+    responses={"200": {"description": "User successfully deleted"}, "404": {"description": "User not found"}},
+    summary="Delete a user",
+    allow_access=True,
+)
+class DeleteUser(BaseUser):
+    async def __call__(self):
+        user = await self.get_user()
+        view = DefaultDELETE(user, self.request)
+        return await view()
+
+
+@configure.service(
+    context=IContainer,
+    name="@users",
+    method="GET",
+    permission="guillotina.ManageUsers",
+    responses={
+        "200": {
+            "description": "List of users",
+            # TODO: add response content schema here
+        }
+    },
+    summary="List existing users",
+    allow_access=True,
+)
+class ManageAvailableUsers(Service):
+    async def __call__(self):
+        users = await self.get_users_form_folder()
+        if not users:
+            return []
+        result = []
+        for user in users:
+            serializer = get_multi_adapter((user, self.request), IResourceSerializeToJsonSummary)
+            result.append(await serializer())
+
+        return result
+
+    async def get_users_form_folder(self):
+        user_folder = await navigate_to(self.context, "users")
+        users = []
+        async for _, user in user_folder.async_items():
```

Comment:
Can we search on the catalog if its available or go on the folder as a fallback ?